### ISSUE TITLE
oa_la_mapping data requires Scotland and Northern Ireland information

### ISSUE THREAD

**Issue by paigeh1** [ONSdigital/SDG_11.2.1#236]
oa_la_mapping data (both 2019 and 2020) does not have Scotland and Ireland output area information, only England and Wales. When we run the script and NI/Scotland local authorities are randomly chosen, the script will fail as it doesn't exist. We have attached the local authorities that this affects.

For this ticket, research will be needed to look into where we can find Scotland and NI data for the output areas so we have all of the UK.

[SDG_11.2.1Unique_LAs.csv](https://github.com/ONSdigital/SDG_11.2.1/files/8901773/SDG_11.2.1Unique_LAs.csv)

**Comment by jwestw** [comment]
Excellent find! This is exactly the reason for randomising the OA selection!

**Comment by james-westwood** [comment]
This has a linked issue: https://github.com/ONSdigital/SDG_11.2.1/issues/67

**Comment by Antonio-John** [comment]
Hi @MusaChirikeni

-Do you know if there is an OA to LA mapping that includes those of Scotland/NI please? Currently we are using just the England&Wales version.

Many thanks,

Antonio

**Comment by MusaChirikeni** [comment]
Hi @Antonio-John,

[Here is the Hierarchy of UK Geographies for reference. Look out for the 2011 Statistical Building Block](https://geoportal.statistics.gov.uk/documents/hierarchical-representation-of-uk-statistical-geographies-december-2021/explore)

There are OAs for GB but these are called [Small Areas](geoportal.statistics.gov.uk/datasets/096a7ccbc8e244cc972189b2f07a321a_0/explore) in NI. 

Something to be aware of also is that mid-year Population Estimates are only released at OA level in England & Wales and [Small Areas in NI](https://www.nisra.gov.uk/statistics/population/population-estimates-small-areas). However, Scotland does not have a yearly release of OA population estimates. There is only a 2011 population for those from the Census.
Yearly population estimates for Scotland are only available from the next Geographical level ([DataZones](https://www.nrscotland.gov.uk/statistics-and-data/statistics/statistics-by-theme/population/population-estimates/mid-year-population-estimates), equivalent to LSOAs),  going up. 

In short, you can do an OA to LAD lookup for the UK but you've got to be aware of the above mentioned complications when working with population estimates data.

best regards,

**Comment by Antonio-John** [comment]
thanks for information :) @MusaChirikeni

**Comment by Antonio-John** [comment]
These are output areas population stats for Scotland 2011 census.
[KS101SC.csv](https://github.com/ONSdigital/SDG_11.2.1/files/9214008/KS101SC.csv)

**Comment by Antonio-John** [comment]
Initial results for 1 LA
[Scotland_results.csv](https://github.com/ONSdigital/SDG_11.2.1/files/9217056/Scotland_results.csv)

**Comment by Antonio-John** [comment]
Northern Ireland - Bus Stop https://www.opendatani.gov.uk/dataset/translink-bus-stop-list/resource/29f3f2fd-d131-4b86-8933-42b5b3763763
Northern Ireland - Train Stops https://www.opendatani.gov.uk/dataset/northern-ireland-railways-stations/resource/967e32c3-1cc2-4aee-b485-92121a32eb4d
Northern Ireland - Pop Estimates for small areas/OA - https://www.nisra.gov.uk/statistics/population/population-estimates-small-areas
Northern Ireland - Population Weighted Centroids for small areas OA https://www.nisra.gov.uk/support/geography/northern-ireland-small-areas
Northern Ireland - small areas to LA lookup ??

**Comment by Antonio-John** [comment]
Steps to be able to run Scottish Data:
You'll need a few extra files, I'll detail which ones and where to save them here.
1) Scottish Usual Resident Population. This is the population per output area from the 2011 census. [KS101SC.csv](https://github.com/ONSdigital/SDG_11.2.1/files/9214008/KS101SC.csv) Needs to be placed in data/
2) Population weighted centroids are found [here  ](https://www.spatialdata.gov.scot/geonetwork/srv/api/records/9d977a73-7884-4870-ae76-afccf8e6fae8) they need to be placed in data/pop_weighted_centroids/scotland/2011/ 
3) [OA to LA lookup](https://geoportal.statistics.gov.uk/datasets/ons::postcode-to-output-area-to-lower-layer-super-output-area-to-middle-layer-super-output-area-to-local-authority-district-november-2021-lookup-in-the-uk/about) needs to be placed in data/oa_la_mapping/scotland/2021

**Comment by jwestw** [comment]
@Antonio-John can we establish how "Usual Resident Population" is different from the population figures? And if we are using it for Scotland, for consistency should we use it for E/W and NI?

**Comment by paigeh1** [comment]
We have pipelines for both Scotland and NI now fully complete and working. Ticket's now complete due to this so I'm going to close it.
